test/upload_test.js

```diff
@@ -27,6 +27,14 @@ describe('S3 Webpack Upload', () => {
         .then(assertFileMatches);
     });
 
+    it('uploads entire directory with alternate pathing s3', () => { // eslint-disable-line
+      config = testHelpers.createWebpackConfigAlt({ s3Config });
+
+      return testHelpers.runWebpackConfig({ config })
+        .then(testHelpers.testForFailFromDirectoryOrGetS3Files(testHelpers.OUTPUT_PATH_ALT))
```

Comment:
I see two things here:
1. You pass `OUTPUT_PATH_ALT`, which is `bundle` directory, but image file is not there. It is in `assets` directory. So this test is not testing entire functionality.
2. Function `testForFailFromDirectoryOrGetS3Files` at the end of the chain invokes `expected: this.readFileFromOutputDir(fetchFile),`, which uses OUTPUT_PATH. That is why test fails - there are no files within `OUTPUT_PATH`. My suggestion is to remove entirely `OUTPUT_PATH_ALT`.

Reply:
Removed OUTPUT_PATH_ALT in favor of the regular OUTPUT_PATH